### PR TITLE
refactor CallStatement, DeleteStatement and DoStatement in DMLStatement

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #7170.

Changes proposed in this pull request:
- refactor CallStatement, DeleteStatement and DoStatement in DMLStatement

### CODE REVIEW

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/impl/ShardingDeleteStatementValidator.java [hunk 1]
```diff
@@ -42,7 +44,7 @@ public void preValidate(final ShardingRule shardingRule, final RouteContext rout
     
     @Override
     public void postValidate(final DeleteStatement sqlStatement, final RouteResult routeResult) {
-        if (sqlStatement.getLimit().isPresent() && routeResult.getRouteUnits().size() > 1) {
+        if (sqlStatement instanceof MySQLStatement && ((MySQLDeleteStatement) sqlStatement).getLimit().isPresent() && routeResult.getRouteUnits().size() > 1) {
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/impl/ShardingDeleteStatementValidator.java:47]: Extracting a function is better.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dml/impl/DeleteStatementAssert.java [hunk 2]
```diff
@@ -49,8 +51,10 @@
     public static void assertIs(final SQLCaseAssertContext assertContext, final DeleteStatement actual, final DeleteStatementTestCase expected) {
         assertTable(assertContext, actual, expected);
         assertWhereClause(assertContext, actual, expected);
-        assertOrderByClause(assertContext, actual, expected);
-        assertLimitClause(assertContext, actual, expected);
+        if (actual instanceof MySQLStatement) {
+            assertOrderByClause(assertContext, (MySQLDeleteStatement) actual, expected);
+            assertLimitClause(assertContext, (MySQLDeleteStatement) actual, expected);    
+        }
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dml/impl/DeleteStatementAssert.java:57]: How about to extract A function named `assertXXXXX`

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dml/impl/DeleteStatementAssert.java:57]: It is an optional item. Just from my glance, my feeling is that it is better to gather this judgment and handling detail into a function.

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/impl/ShardingDeleteStatementValidator.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/impl/ShardingDeleteStatementValidator.java:47]: > Extracting a function is better.

@tristaZero I will optimize it.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dml/impl/DeleteStatementAssert.java (hunk 2 above)
Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dml/impl/DeleteStatementAssert.java:57]: > It is an optional item. Just from my glance, my feeling is that it is better to gather this judgment and handling detail into a function.

It is a good idea. 👍

**Review by tristaZero — APPROVED** [review]
Nice!

**Pull request merged**